picodom.d.ts

```diff
@@ -1,23 +1,38 @@
-export interface VNode<Data> {
+export as namespace picodom
+
+export interface VNode<Props> {
   type: string
   props?: Props
-  children: VNode<Props> | string
+  children: Array<VNode<{}> | string>
 }
 
-export interface Component<Props> {
-  (props?: Props, ...children: VNode<Props> | string): VNode<Props>
+export interface Component<Props = {}> {
+  (props: Props, children: Array<VNode<{}> | string>): VNode<Props>
```

Comment:
Instead of making props optional, I defined the default type in the generic type param since a Component function should never received `undefined` as props (re: [line 23 of h.js](https://github.com/picodom/picodom/blob/d0c95dd50678d6df220e8e642e5b7365e622bc2b/src/h.js#L23))